specs/predeploys.md

```diff
@@ -93,290 +118,176 @@ In the Bedrock system, this contract will no longer be used as part of the
 
-    function deposit() public;
-    function withdraw(uint wad) public;
-    function totalSupply() public view returns (uint);
-    function approve(address guy, uint wad) public returns (bool);
-    function transfer(address dst, uint wad) public returns (bool);
+Address: `0x4200000000000000000000000000000000000006`
 
-    function transferFrom(
-        address src,
-        address dst,
-        uint wad
-    ) public returns (bool);
-}
-```
+`WETH9` is the standard implementation of Wrapped Ether on Optimism. It is a
+commonly used contract and is placed as a predeploy so that it is at a
+deterministic address across chains.
 
 ## L2CrossDomainMessenger
 
-The `L2CrossDomainMessenger` is used to give a better user experience when
-sending cross domain messages from L2 to L1. It extends the
-`CrossDomainMessenger`, which allows for replayability of messages. Any calls
-through the `l1CrossDomainMessenger` go through the `L2CrossDomainMessenger`.
+[Implementation](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L2CrossDomainMessenger.sol)
+
+Address: `0x4200000000000000000000000000000000000007`
+
+The `L2CrossDomainMessenger` gives a higher level API for sending cross domain
+messages. It maintains a mapping of L1 messages that have been relayed to L2
+to prevent replay attacks and also allows for replayability if the L1 to L2
+transaction reverts on L2.
+
+Any calls to the `L1CrossDomainMessenger` on L1 are serialized such that they
+go through the `L2CrossDomainMessenger` on L2.
 
 The `relayMessage` function executes a transaction from the remote domain while
 the `sendMessage` function sends a transaction to be executed on the remote
 domain through the remote domain's `relayMessage` function.
 
 ## L2StandardBridge
 
-The `L2StandardBridge` is part of the ERC20 and ETH bridging system.
-Users can send ERC20s or ETH to the `L1StandardBridge` and receive the asset on
-L1 through the `L2StandardBridge`. Users can also withdraw their assets through the
-`L2StandardBridge` to L1.
+[Implementation](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L2/L2StandardBridge.sol)
+
+Address: `0x4200000000000000000000000000000000000010`
+
+The `L2StandardBridge` is a higher level API built on top of the
+`L2CrossDomainMessenger` that gives a standard interface for sending ETH or
+ERC20 tokens across domains.
+
+To deposit a token from L1 to L2, the `L1StandardBridge` locks the token and
+sends a cross domain message to the `L2StandardBridge` which then mints the
+token to the specified account.
+
+To withdraw a token from L2 to L1, the user will burn the token on L2 and the
+`L2StandardBridge` will send a message to the `L1StandardBridge` which will
+unlock the underlying token and transfer it to the specified account.
 
 ## SequencerFeeVault
```

Comment:
Should we move this section so that it's next to the BaseFeeVault and L1FeeVault?

Reply:
Sure, was trying to follow the same order as in the table but i'm not opposed to ordering by functionality below the table